extensions/wikia/WikiFactory/Dumps/DumpsOnDemand.php

```diff
@@ -116,45 +115,44 @@ static public function sendMail( $sDbName = null, $iCityId = null, $bHidden = fa
 	 * @access public
 	 */
 	static public function queueDump( $iCityId = null, $bHidden = false, $bClose = false ) {
-            
-            if ( is_null( $iCityId ) ) {
-                global $wgCityId;
-                $iCityId = $wgCityId;
-            }
-            
-            $oWiki = WikiFactory::getWikiByID( $iCityId );
-            
-            if ( !is_object( $oWiki ) ) {
-                trigger_error( sprintf( '%s terminated. No such wiki (city_id: %d.', __METHOD__, $iCityId ) , E_USER_WARNING );
-                return null;
-            }
-            
-            global $wgUser;
-            
-            $aData = array(
-                'dump_wiki_id'      => $iCityId,
-                'dump_wiki_dbname'  => $oWiki->city_dbname,
-                'dump_wiki_url'     => $oWiki->city_url,
-                'dump_user_name'    => $wgUser->getName(),
-                'dump_requested'    => wfTimestampNow()
-            );
-            
-            if ( $bHidden ) {
-                $aData['dump_hidden'] = 'Y';
-            }
-            
-            if ( $bClose ) {
-                $aData['dump_closed'] = 'Y';
-            }
-            
-            $oDB = wfGetDB( DB_MASTER, array(), 'wikicities' );
-            $oDB->insert( 'dumps', $aData, __METHOD__ );
-            $oDB->update(
-                    'city_list',
-                    array( 'city_lastdump_timestamp' => wfTimestampNow() ),
-                    array( 'city_id' => $iCityId ),
-                    __METHOD__
-            );
+		global $wgCityId, $wgUser;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)